resources/design-document.md

```diff
@@ -0,0 +1,335 @@
+* Retrieve translation client information
+* Retrieve a list of all translation clients, sorted alphabetically
+* Retrieve a list of all translation clients, sorted by most recent activity
+* Retrieve all translation cases associated with a translation client
+* Update information about the translation client
+* Archive (soft-delete) a translation client
+
+### Stretch Goals
+
+#### Estimates
+* Create a new estimate
+* Save an estimate by assigning it to an existing translation client or by creating a new translation client
+* Retrieve past estimates
+* Retrieve a list of all estimates, sorted by project type
+* Retrieve a list of all estimates, sorted by translation client
+* Link an existing estimate to a new translation case
+* Update the estimate with notes on negotiation or other details (saved estimate is read-only, notes can be edited)
+* Archive (soft-delete) an estimate
+
+#### Statistics
+* Pull statistics from the translation cases
+  * Words translated per hour compared by project type
+  * Most productive time of day for each project type
+  * Largest project type by case count
+  * Total translation time since beginning of year / month
+  * Translation clients with the fastest payment turnaround time
+  * etc.
+
+
+### 4.2. Out of Scope
+
+* Linking accounts
+* Storing translated documents
+* Making translation keywords searchable
+* Creating estimates that can be sent to customers directly
+
+# 5. Proposed Architecture Overview
+
+This initial iteration will provide the minimum loveable product (MLP) including creating, retrieving, updating and 
+  archiving translation cases and translation clients, as well as sorting these objects in useful ways.
+
+I will use API Gateway and Lambda to create seven endpoints (`GetTranslationCase`, `CreateTranslationCase`, 
+`UpdateTranslationCase`, `ArchiveTranslationCase`, `GetTranslationClient`, `UpdateTranslationClient`, 
+  `ArchiveTranslationClient`) that will handle the creation, update, retrieval and archiving of translation clients 
+  and translation cases.
+
+I will store the translation clients and translation cases in tables in DynamoDB.
+
+TranslationTracker will also provide a web interface for users to manage their translation cases and translation 
+  clients. A main page providing lists of the translation cases with the soonest upcoming deadlines and translation 
+  clients with most recent activity will allow creation of new translation cases and link off to pages per-translation 
+  client and per-translation case to update translation progress and other details.
+
+# 6. API
+
+## 6.1. Public Models
+
+```
+// TranslationCaseModel
+
+String translationCaseId;
+String translationClientId;
+String sourceTextTitle;
+String sourceTextAuthor;
+String translatedTitle;
+String caseNickname;
+Double rate;
+String rateUnit;
+Integer count;
+String countUnit;
+Double grossPayment;
+Double taxRate;
+String projectType;
```

Comment:
Would this be a good candidate to be an Enum? (Are there a set list of project types, or would this be free-form?)

Reply:
They are good candidates, I have made this and the translationClientType both enums